### DISCUSSION TITLE
Need help attaching my JavaScript code to be displayed on the _projects page

### DISCUSSION THREAD

**Discussion by DamilolaOwolabi** [alshedivat/al-folio#2635] [Q&A] [answered]
Hi all, thank you for the amazing work.  What you have done is impressive, and I appreciate you making it open-source. I have completed some d3.js and p5js projects during the previous school semester. I need advice/suggestions on how to attach the code to the _projects page.

I understand the code used is a markdown code, But I'm not that experienced with GitHub pages and the like. I have tried using different methods, especially those suggested by ChatGPT to attach my JS code to the page, but none of them worked. Please I would appreciate any advice or any example code on how to resolve this issue. Thank you for reading up to this point.

**Comment by george-gca** [comment, marked as answer]
I don't believe ChatGPT would give a useful answer in this case, at least not without enough context.

For instance, this template is made using `jekyll`. Jekyll uses markdown, yes, but down below it compiles to html. You can even use html inside your markdown files. We actually do that in the [projects](https://github.com/alshedivat/al-folio/blob/master/_pages/projects.md) page. So, if you want to use these libraries in a single project page, you can add it to the project markdown itself.

About the workings of it, I recommend reading jekyll docs pages about how it creates a page ([1](https://jekyllrb.com/docs/layouts/) and [2](https://jekyllrb.com/docs/step-by-step/04-layouts/)). In our case a [project](https://github.com/alshedivat/al-folio/blob/master/_projects/1_project.md) have a layout to it (which is [page](https://github.com/alshedivat/al-folio/blob/master/_layouts/page.liquid) that inherits from [default](https://github.com/alshedivat/al-folio/blob/master/_layouts/default.liquid)).

Now, if you want to add these libraries to more pages, the best way for you to do that is to follow something like what is done for some blog posts. Check, for example, the [echarts](https://github.com/alshedivat/al-folio/blob/master/_posts/2024-01-26-echarts.md) post, and how it is built. See what is defined in the page front matter, where these variables are used and how.

We add `d3.js` in our template directly, but only for [mermaid diagram zooming support](https://github.com/alshedivat/al-folio/blob/3ff7579a7419fc546816535361a8b90c7c49553d/_includes/scripts/mermaid.liquid#L11).
